packages/expect/src/asymmetric_matchers.js

```diff
@@ -15,12 +15,18 @@ import {
   isUndefined,
 } from './jasmine_utils';
 
+import {emptyObject} from './utils';
+
 class AsymmetricMatcher {
   $$typeof: Symbol;
 
   constructor() {
     this.$$typeof = Symbol.for('jest.asymmetricMatcher');
   }
+
+  toString() {
+    return this.constructor.name;
```

Comment:
Can you revert this? I do not want us to rely on  class names in JavaScript. We may want to minify or mangle these at some point and this would break. Any user visible value should come from strings, not from data structure reflection.